### PR TITLE
Add Support for RHEL 8 and 9

### PR DESCRIPTION
**Pull request by timothylytle**
IN PROGRESS -  still testing github actions

### CODE REVIEW

File: tasks/install_rhel.yml [hunk 1]
```diff
@@ -4,35 +4,41 @@
...
 
-      - name: PostgreSQL | Make sure the dependencies are installed | yum
-        yum:
-         name: ["python-psycopg2", "python-pycurl", "glibc-common","libselinux-python"]
-         state: present
-         update_cache: yes
+      - name: PostgreSQL | Disable postgresql module (necessary for RHEL8+)
+        command:
+          cmd: dnf module disable postgresql -y
+        when: "ansible_distribution_major_version == '8' or ansible_distribution_major_version == '9'"
```
Comment (genben) [inline tasks/install_rhel.yml]: I think, a condition should be >= 8.
```
ansible_distribution_major_version is version(‘>=‘, ‘8’)
```

File: tasks/install_rhel.yml [hunk 2]
```diff
@@ -4,35 +4,41 @@
...
            gpgkey: "{{ postgresql_yum_repository_gpgkey }}"
+           enabled: yes
         when: postgresql_install_repository
 
-      - name: PostgreSQL | Make sure the dependencies are installed | yum
-        yum:
-         name: ["python-psycopg2", "python-pycurl", "glibc-common","libselinux-python"]
-         state: present
-         update_cache: yes
+      - name: PostgreSQL | Disable postgresql module (necessary for RHEL8+)
```
Comment (genben) [inline tasks/install_rhel.yml:25]: Please add in comments why wit must be disabled

File: tasks/main.yml [hunk 3]
```diff
@@ -19,11 +20,11 @@
   when: ansible_pkg_mgr == "apt"
   tags: [postgresql, postgresql-install]
 
-- import_tasks: install_yum.yml
-  when: ansible_pkg_mgr == "yum" and ( ansible_distribution == "RedHat" or ansible_distribution == "CentOS" or ansible_distribution == "OracleLinux")
+- import_tasks: install_rhel.yml
+  when: (ansible_pkg_mgr == "yum" or ansible_pkg_mgr == "dnf") and (ansible_distribution == "RedHat" or ansible_distribution == "CentOS" or ansible_distribution == "OracleLinux" or ansible_distribution == "Rocky")
```
Comment (genben) [inline tasks/main.yml]: The condition can be shorter:
When os family is redhat and distro is not fedora

File: tasks/main.yml [hunk 4]
```diff
@@ -19,11 +20,11 @@
...
   tags: [postgresql, postgresql-install]
 
-- import_tasks: install_dnf.yml
+- import_tasks: install_fedora.yml
```
Comment (genben) [inline tasks/main.yml:26]: Condition here can be just “when distro is fedora”.
Fedora has migrated to dnf many years ago. 
There is no yum on Fedora now.

File: vars/RedHat.yml [hunk 5]
```diff
@@ -18,3 +18,11 @@ postgresql_unix_socket_directories:
 
 postgresql_fdw_mysql_packages: "mysql_fdw_{{ postgresql_version_terse }}"
 postgresql_fdw_ogr_packages: "ogr_fdw{{ postgresql_version_terse }}"
+
+postgresql_packages:
+  - ca-certificates
+  - python-psycopg2
+  - python-pycurl
+  - glibc-common
+  - epel-release
```
Comment (genben) [inline vars/RedHat.yml]: I think, the setup of packages must be done in to steps:
1. Enable EPEL repository
2. Install the packages, using EPEL repository

File: vars/RedHat_8.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+postgresql_fdw_mysql_packages: "mysql_fdw_{{ postgresql_version_terse }}"
+postgresql_fdw_ogr_packages: "ogr_fdw{{ postgresql_version_terse }}"
+
+postgresql_packages:
+  - ca-certificates
+  - python3-psycopg2
+  - python3-pycurl
+  - glibc-common
+  - epel-release
```
Comment (genben) [inline vars/RedHat_8.yml]: Please verify if EPEL is required for RockyLinux 8,9.
Maybe, it is not required.

File: .ansible-lint [hunk 7]
```diff
@@ -0,0 +1,10 @@
+---
+exclude_paths:
+  - molecule/
+  - tests/
+
+skip_list:
+ - fqcn
+ - yaml[line-length]
+ - yaml[empty-lines]
+ - yaml[trailing spaces]
```
Comment (genben) [inline .ansible-lint]: Please add to ignore list

- yaml[truthy]
- jinja[spacing]

File: tasks/install_rhel.yml (hunk 1 above)
Comment (timothylytle) [reply tasks/install_rhel.yml]: This has been updated

File: tasks/main.yml (hunk 3 above)
Comment (timothylytle) [reply tasks/main.yml]: This has been updated

File: tasks/install_rhel.yml (hunk 2 above)
Comment (timothylytle) [reply tasks/install_rhel.yml:25]: I added a comment and a link,  NOTE there is an alternative way to fix this issue, by updating the `.repo` file, this way was the method chosen by the upstream repo

File: tasks/main.yml (hunk 4 above)
Comment (timothylytle) [reply tasks/main.yml:26]: This has been updated

File: vars/RedHat_8.yml (hunk 6 above)
Comment (timothylytle) [reply vars/RedHat_8.yml]: It was not required, I have removed from the list

File: .ansible-lint (hunk 7 above)
Comment (timothylytle) [reply .ansible-lint]: This has been updated, I also corrected the typo in `yaml[trailing spaces]` > `yaml[trailing-spaces]`

File: vars/RedHat.yml (hunk 5 above)
Comment (timothylytle) [reply vars/RedHat.yml]: @genben the way I understand it, Just the postgresql packages, like `postgresql12-server`, `postgesl112`, and `postgresql12-contrib` require the EPEL repository, those packages are installed on a later step.  

The only change here was moving the list of dependencies to a variable
```yaml
      - name: PostgreSQL | Install all the required dependencies | yum
        yum:
          name: ["ca-certificates","python-psycopg2", "python-pycurl", "glibc-common","epel-release","libselinux-python"]
          state: present
```
was changed to:
```yaml
      - name: PostgreSQL | Install all the required dependencies (yum) | RHEL
        yum:
          name: "{{ postgresql_packages }}"
          state: present
          update_cache: yes
```

A more descriptive name for the variables would be  `postgresql_packages_dependencies`, but i was trying to follow the upstream repo

**Pull request closed without merging**